app/controllers/support/providers/users_check_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Support
+  module Providers
+    class UsersCheckController < SupportController
+      def show
+        @user_form = UserForm.new(current_user, user)
+      end
+
+      def update
+        @user_form = UserForm.new(current_user, user)
+        if @user_form.save!
+          UserAssociationsService::Create.call(user: @user_form.model, provider:) if @user_form.model.providers.exclude?(provider)
+          redirect_to support_provider_users_path
+          flash[:success] = "User added"
+        end
+      end
+
+    private
+
+      def user
+        provider
```

Comment:
I think you can just call `provider` in `show` so it clearly shows that you're prepping the variable for view. If I were to call this method for example, I'd expect it to give me a user without some additional side effect of preparing a provider (unless the method name described the intent)

Reply:
Totally agree that this is much better! Thanks! 

Done in [4e711dad1e75e1b291f2fd99fdb26f37183caf4c](https://github.com/DFE-Digital/publish-teacher-training/pull/2839/commits/4e711dad1e75e1b291f2fd99fdb26f37183caf4c)